lib/index.js

```diff
@@ -39,44 +39,75 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
   const accountPath = basePath + '/:account';
   const {baseUri} = bedrock.config.server;
 
+  /**
+   * @function createNewAccount
+   * @param {Request} req
+   * @param {Response} res
+   * @description
+   */
+  async function createNewAccount(req, res) {
+    const account = {
+      id: 'urn:uuid:' + uuid(),
+      email: req.body.email
+    };
+    if(!account.email) {
+      throw new BedrockError(
+        'No Email', 'InvalidEmailError', {
+          httpStatusCode: 400,
+          public: true,
+        });
+    }
+    const meta = {
+      sysResourceRole: [{
+        sysRole: 'account.registered',
+        resource: [account.id]
+      }]
+    };
+      // actor is not required, but passed through if given
+    const {actor = null} = req.user || {};
+    await brAccount.insert({actor, account, meta});
+    res.status(201).location(baseUri + '/' + account.id).json(account);
+  }
+
   app.post(
     basePath,
     optionallyAuthenticated,
     // TODO: validate('services.account.postAccounts')
-    asyncHandler(async (req, res) => {
-      const account = {
-        id: 'urn:uuid:' + uuid(),
-        email: req.body.email
-      };
-      const meta = {
-        sysResourceRole: [{
-          sysRole: 'account.registered',
-          resource: [account.id]
-        }]
-      };
-      // actor is not required, but passed through if given
-      const {actor = null} = req.user || {};
-      await brAccount.insert({actor, account, meta});
-      res.status(201).location(baseUri + '/' + account.id).json(account);
-    }));
-
+    asyncHandler(createNewAccount));
+  /**
```

Comment:
Needs empty line of whitespace to break this up (though the function below is a private helper so it should move to the bottom of the file anyway).

Reply:
we could actually separate the actual handlers into their own module and then call this routes we have yet to implement delete and patch for instance so doing it now would mean when those handlers are added we expand in another file. separate routes from handler.

Reply:
Own module or own file?

Reply:
I just want to make sure we've got the right organizational balance here ... if we're creating too few files it's a mess, if we're creating too many it's too hard to navigate/compare/find what you want. We certainly don't want another module... but a better file structure might be ok.

cc: @davidlehn, @mattcollier

Reply:
I am actually am voting for a separate file that we import from. looking at the file it's just kind of complicated. we have several event emitter callbacks and then several different middleware compositions. if we move the handler functions to a different file it means routes will clarify which middleware goes where etc. and handlers can be found more easily.

Reply:
this the only remaining task so I will let it say at the bottom of the current module for now.